Code/Tools/AssetProcessor/native/utilities/BuilderManager.cpp

```diff
@@ -220,6 +220,20 @@ namespace AssetProcessor
         #endif // !AZ_TRAIT_OS_PLATFORM_APPLE && !AZ_TRAIT_OS_USE_WINDOWS_FILE_PATHS
         }
 
+        auto settingsRegistry = AZ::SettingsRegistry::Get();
+        bool skipAtomOutput = false;
+        if (settingsRegistry && settingsRegistry->Get(skipAtomOutput, "/O3DE/SceneAPI/AssetImporter/SkipAtomOutput") && skipAtomOutput)
+        {
+            const char* settingString =
+#if !AZ_TRAIT_OS_PLATFORM_APPLE && !AZ_TRAIT_OS_USE_WINDOWS_FILE_PATHS
+                R"( --regset="/O3DE/SceneAPI/AssetImporter/SkipAtomOutput=true")";
+#else
+                R"( --regset="\"/O3DE/SceneAPI/AssetImporter/SkipAtomOutput=true\"")";
+#endif
+
+            params.append(settingString);
+        }
+
         return params;
```

Comment:
It really doesn't feel like mentioning atom is the right thing to do in the Builder Manager.

If you have specified an actual O3DE/SceneAPI/AssetImporter/SkipAtomOutput registry set via registry file, this would automatically propogate to children since they also read the registry.  Why is this necessary then?

MAybe theres a way for the script to prepare the reg-set file for the test, or maybe there's a way to not have to actually do this to get the desired outcome.   Its hard to tell from the code, but it looks like setting this to true just causes BindToCall(&ModelExporterComponent::ExportModel); to not happen.  Whats the actual outcome?

Reply:
I'm trying to avoid writing anything to disk because it seems to be an endless source of flakiness in testing.  I should add a comment in the code to explain the point of this:

Atom's material and mesh processors pull in a bunch of external dependencies when processing an FBX file.  I looked into the material processor and it pulls in something like 73 dependent files that need to be processed in order to successfully process an FBX with a material on it.

These python tests are set up to process just the files we're testing in a temp directory so they can run really fast.  Having to find some way to pull in and process all those extra dependencies (which could change at any time) felt like the wrong way to go.  The tests just need to inspect the dbgsg file, so there's no point in generating material and mesh outputs. By avoiding the BindToCall, the material and mesh outputs are disabled (to be clear, it's not the output that matters, it's the processing to generate the output which causes errors).

I called it SkipAtomOutput to make it a bit more generic, instead of SkipMeshAndMaterialOutput, in case in the future there's some 3rd component we need to disable.  Maybe SkipRenderOutput is better?  Also it wasn't an option to disable all the Exporters because there is a test that checks the physics soft naming rules for physx output, so that Exporter needs to remain enabled.

Reply:
How capturing any regset/regremove options from the CommandLine and forward those to the AssetBuiilder
```c++
auto settingsRegistry = AZ::SettingsRegistry::Get();
AZ::CommandLine commandLine;
GetCommandLineFromRegistry(settingsRegisty, commandLine);
AZStd::fixed_vector optionKeys{ "regset", "regremove" );
for (auto&& optionKey : optionKeys)
{
    size_t commandOptionCount = commandLine.GetNumSwitchValues(optionKey);
    for (size_t optionIndex = 0; optionIndex < commandOptionCount; ++optionIndex)
    {
        const AZStd::string& optionValue = commandLine.GetSwitchValue(optionKey, optionIndex);
        params.append(AZStd::string::format(R"(--%s="%s")", optionKey, optionValue.c_str()));
    }
}
```

Reply:
That works great, updated